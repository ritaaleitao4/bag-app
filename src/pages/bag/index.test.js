import mockDatabase from '../../../server/database';
import BagService from '../../services/bag-service';
import BagPage from '.';

jest.mock('../../services/bag-service');
const bagId = 'bag1';

describe('bag page', () => {
    beforeEach(() => jest.clearAllMocks());

    it('it renders correctly', async () => {
        BagService
            .getBagByIdAsync
            .mockImplementationOnce(() => Promise.resolve(mockDatabase.bag[0]));

        expect(await BagPage.load(bagId)).toMatchSnapshot();
    });

    it('it renders differently when there is an error', async () => {
        BagService
            .getBagByIdAsync
            .mockImplementationOnce(() => Promise.reject());

        expect(await BagPage.load(bagId)).toMatchSnapshot();
    });
});
