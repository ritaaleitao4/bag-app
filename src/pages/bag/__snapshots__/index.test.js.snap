// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bag page it renders correctly 1`] = `

<div class="page">
  <main class="product-list alignCenter">
    <div class="product-card">
      <div class="remove-btn">
        <img src="test-file-stub">
      </div>
      <div class="image-container">
        <img class="image"
             src="https://via.placeholder.com/130x175"
        >
      </div>
      <div class="info-container alignCenter">
        <div class="brandName">
          BRAND 1
        </div>
        <div class="price">
          €100.40
        </div>
      </div>
    </div>
    <div class="product-card">
      <div class="remove-btn">
        <img src="test-file-stub">
      </div>
      <div class="image-container">
        <img class="image"
             src="https://via.placeholder.com/130x175"
        >
      </div>
      <div class="info-container alignCenter">
        <div class="brandName">
          BRAND 2
        </div>
        <div class="price">
          €100.20
        </div>
      </div>
    </div>
    <div class="product-card">
      <div class="remove-btn">
        <img src="test-file-stub">
      </div>
      <div class="image-container">
        <img class="image"
             src="https://via.placeholder.com/130x175"
        >
      </div>
      <div class="info-container alignCenter">
        <div class="brandName">
          BRAND 3
        </div>
        <div class="price">
          €150.33
        </div>
      </div>
    </div>
    <div class="product-card">
      <div class="remove-btn">
        <img src="test-file-stub">
      </div>
      <div class="image-container">
        <img class="image"
             src="https://via.placeholder.com/130x175"
        >
      </div>
      <div class="info-container alignCenter">
        <div class="brandName">
          BRAND 4
        </div>
        <div class="price">
          €150.33
        </div>
      </div>
    </div>
    <div class="product-card">
      <div class="remove-btn">
        <img src="test-file-stub">
      </div>
      <div class="image-container">
        <img class="image"
             src="https://via.placeholder.com/130x175"
        >
      </div>
      <div class="info-container alignCenter">
        <div class="brandName">
          BRAND 5
        </div>
        <div class="price">
          €50.21
        </div>
      </div>
    </div>
    <div class="product-card">
      <div class="remove-btn">
        <img src="test-file-stub">
      </div>
      <div class="image-container">
        <img class="image"
             src="https://via.placeholder.com/130x175"
        >
      </div>
      <div class="info-container alignCenter">
        <div class="brandName">
          BRAND 6
        </div>
        <div class="price">
          €230.39
        </div>
      </div>
    </div>
    <div class="product-card">
      <div class="remove-btn">
        <img src="test-file-stub">
      </div>
      <div class="image-container">
        <img class="image"
             src="https://via.placeholder.com/130x175"
        >
      </div>
      <div class="info-container alignCenter">
        <div class="brandName">
          BRAND 7
        </div>
        <div class="price">
          €1030.99
        </div>
      </div>
    </div>
    <div class="product-card">
      <div class="remove-btn">
        <img src="test-file-stub">
      </div>
      <div class="image-container">
        <img class="image"
             src="https://via.placeholder.com/130x175"
        >
      </div>
      <div class="info-container alignCenter">
        <div class="brandName">
          BRAND 8
        </div>
        <div class="price">
          €300.30
        </div>
      </div>
    </div>
    <div class="product-card">
      <div class="remove-btn">
        <img src="test-file-stub">
      </div>
      <div class="image-container">
        <img class="image"
             src="https://via.placeholder.com/130x175"
        >
      </div>
      <div class="info-container alignCenter">
        <div class="brandName">
          BRAND 9
        </div>
        <div class="price">
          €130.90
        </div>
      </div>
    </div>
  </main>
  <footer class="footer alignCenter">
    @ Copyright E-commerce Store
  </footer>
</div>

`;

exports[`bag page it renders differently when there is an error 1`] = `

<div class="error">
  Error loading bag items...
</div>

`;
